selftests/functional/test_test.py

```diff
@@ -0,0 +1,69 @@
+import json
+import os
+import shutil
+import tempfile
+import unittest
+
+from avocado.core import exit_codes
+from avocado.utils import process
+from avocado.utils import script
+
+basedir = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..')
+basedir = os.path.abspath(basedir)
+
+AVOCADO = os.environ.get("UNITTEST_AVOCADO_CMD", "./scripts/avocado")
+
+TEST_SETUP_EXCEPTION = """
+import avocado
+
+class AvocadoTest(avocado.Test):
+
+    def setUp(self):
+        self.log.info('setup code before')
+        raise
```

Comment:
You mean something like this (WiP)?
https://github.com/apahim/avocado/commit/ff23ccb2b6ad0d53067249a9379698ebd7fa910e